### PR TITLE
11 implement osi validation check against a set of rules

### LINKED ISSUE #11: Implement osi-validation check against a set of rules
**Is your feature request related to a problem? Please describe.**
It is currently not checked, if the output messages specified in the readme are actually filled by the model.

**Describe the solution you'd like**
- Use [osi-validation](https://github.com/OpenSimulationInterface/osi-validation) rules to defined the output messages currently specified in the readme.
- Use osi-validation in the CI pipeline to check the model output.

### PR DESCRIPTION
**Pull request by ClemensLinnhoff**
**Reference to a related issue in the repository**
#11, #14

**Add a description**
- [x] Add test case writing model outputs to trace files
- [x] Add osi-validation to the pipeline
- [x] Add osi validation rules for the model output
- [x] Document how to use rules and osi-validation in OpenMSL: see https://github.com/openMSL/doc/pull/12

**Take this checklist as orientation for yourself, if this PR is ready for Maintainer Review**
- [x] My suggestion follows the [governance rules](https://openmsl.github.io/doc/OpenMSL/organization/governance_rules.html).
- [x] All commits of this PR are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [x] My changes generate no errors when passing CI tests. 
- [x] I updated all documentation (readmes incl. figures) according to my changes.
- [x] I have successfully implemented and tested my fix/feature locally.
- [x] Appropriate reviewer(s) are assigned.

### CODE REVIEW

**Comment by ClemensLinnhoff** [comment]
OSI-Validation is now running on both output trace files (TrafficUpdate and TrafficCommandUpdate).

And for both trace files, two sets of rules are tested: The default rules from OSI and custom rules placed in rules/output_rules according to the defined output fields in the readme. This is the output in the [pipeline](https://github.com/openMSL/sl-2-0-traffic-participant-model-repository-template/actions/runs/9270071390/job/25502333842?pr=12):
image

The first test with the default rules does not generate any warning, so everything is compliant with the OSI standard.

The second test generates one warning per trace file:

1. The dismissed_action in the TrafficCommandUpdate is not set, which is correct, because it is only set, when an action is actually dismissed. I am currently not sure how to handle this. So the is_set rule needs to be removed from the rules but we still want to indicate, that this field is filled if necessary.
2. BaseMoving.orientation is not set in the TrafficUpdate. This is actually a bug, that is correctly caught and should be fixed.

The pipeline is currently not failing. I could check if the last line of the individual osi-validation output ends with "--------------" to detect, if there were no warnings and if not set the job to fail.

What do you think @jdsika ?

**Comment by ClemensLinnhoff** [comment]
Pipeline now fails if errors are found in a trace file, as implemented in https://github.com/OpenSimulationInterface/osi-validation/pull/76

File: .github/workflows/build_osi-validation.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        if: steps.cache-osi-validation.outputs.cache-hit != 'true'
+        working-directory: /tmp
+        env:
+          GIT_CLONE_PROTECTION_ACTIVE: false
+        run: git clone https://github.com/OpenSimulationInterface/osi-validation.git
+
+      - name: Checkout develop Branch
+        if: steps.cache-osi-validation.outputs.cache-hit != 'true'
+        working-directory: /tmp/osi-validation
+        run: git checkout develop_traffic_participant_model
```
Comment (ClemensLinnhoff) [inline .github/workflows/build_osi-validation.yml]: This needs to be changed to the main branch, once https://github.com/OpenSimulationInterface/osi-validation/pull/76 and https://github.com/OpenSimulationInterface/osi-validation/pull/74 are merged in osi-validation.

Comment (ClemensLinnhoff) [reply .github/workflows/build_osi-validation.yml]: Everything else is ready for review @jdsika.

**Review by jdsika — APPROVED** [review]

**Pull request merged**